shjq/__init__.py

```diff
@@ -0,0 +1,88 @@
+#!/usr/bin/env python3
+#
+#########################################################################
+#  Copyright 2019 Torsten Dreyer torsten (at) t3r (dot) de
+#  Version 0.0.1
+#########################################################################
+#
+#  This file is part of SmartHomeNG.
+#
+#  SmartHomeNG is free software: you can redistribute it and/or modify
+#  it under the terms of the GNU General Public License as published by
+#  the Free Software Foundation, either version 3 of the License, or
+#  (at your option) any later version.
+#
+#  SmartHomeNG is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU General Public License for more details.
+#
+#  You should have received a copy of the GNU General Public License
+#  along with SmartHomeNG. If not, see <http://www.gnu.org/licenses/>.
+#
+#########################################################################
+
+import logging
+import requests
+from requests_file import FileAdapter
+import pyjq
+from lib.model.smartplugin import SmartPlugin
+
+
+class SHJQ(SmartPlugin):
+    ALLOW_MULTIINSTANCE = True
+    PLUGIN_VERSION = "0.0.1"
+
+    def __init__(self, sh, *args, **kwargs):
+        """
+        Initializes the plugin
+        @param url: URL of the json data to fetch
+        @param cycle: the polling interval in seconds
+        """
+        self.logger = logging.getLogger(__name__)
+        self._url = self.get_parameter_value('url')
+        self._cycle = int(self.get_parameter_value('cycle'))
```

Comment:
Hint: Since **cycle** is defined as int in plugin.yaml, get_parameter_value returns an int. You don't need to convert it any more.